src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1385,7 +1400,12 @@ static void uct_ud_ep_send_ack(uct_ud_iface_t *iface, uct_ud_ep_t *ep)
         goto out;
     }
 
-    if (sizeof(uct_ud_neth_t) <= iface->config.max_inline) {
+    if (uct_ud_ep_ctl_op_check(ep, UCT_UD_EP_OP_SIGNALED)) {
```

Comment:
adding another flag UCT_UD_EP_OP_SIGNALED is not better than checking UCT_UD_EP_FLAG_FLUSH_CANCEL, since both are ep's state flags.
so AFAIU it's not possible to do it per packet, since it can happen from progress context and not uct_ep_flush context, so better revert to checking UCT_UD_EP_FLAG_FLUSH_CANCEL and remove UCT_UD_EP_OP_SIGNALED

Reply:
but we decided to do SIGNALED send operations for both flush(LOCAL) and flush(CANCEL)

Reply:
so can we remove UCT_UD_EP_FLAG_FLUSH_CANCEL, or replace it by UCT_UD_EP_FLAG_SIG_ALL?
since avoiding doing uct_ud_iface_dispatch_async_comps+uct_ud_ep_purge in uct_ud_ep_flush is not really needed, these operations are idempotent

Reply:
(unreslolved)

Reply:
`UCT_UD_EP_FLAG_FLUSH_CANCEL` was removed